Source/DafnyCore/Dafny.atg

```diff
@@ -125,7 +126,8 @@ void CheckDeclModifiers(ref DeclModifierData dmod, string declCaption, AllowedDe
 
 void CheckAttribute(Errors errors, IToken attr) {
   if (attr.val == "dllimport" || attr.val == "handle") {
-    errors.Deprecated(attr, $"Attribute :{attr.val} is deprecated and will be removed in Dafny 4.0");
+    errors.Deprecated(ErrorID.p_deprecated_attribute, Errors.CodeAction(attr, attr.Prev.col, attr.Next.Next.col - attr.Prev.col),
```

Comment:
I wish developers don't need to perform arithmetic at the calling site. Could this be a call like `(attr, attr.Prev, attr.Next.Next)` ?
Also, did these attributes have arguments? If so, we might want to replace the argument IToken attr by the range of the captured attribute so that you don't have an unintuitive `Next..Next` there

Reply:
These did not have attributes. Once Remy has ranges for AST nodes, I can grab the range for the whole Attribute, not just for the id. So I'll let this refactoring wait until the next PR, if you don't object.